disperser/apiserver/server.go

```diff
@@ -636,28 +594,85 @@ func getResponseStatus(status disperser.BlobStatus) pb.BlobStatus {
 	}
 }
 
-func getBlobFromRequest(req *pb.DisperseBlobRequest) *core.Blob {
-	params := make([]*core.SecurityParam, len(req.SecurityParams))
+func (s *DispersalServer) validateRequestAndGetBlob(ctx context.Context, req *pb.DisperseBlobRequest) (*core.Blob, error) {
 
-	for i, param := range req.GetSecurityParams() {
+	data := req.GetData()
+	blobSize := len(data)
+	// The blob size in bytes must be in range [1, maxBlobSize].
+	if blobSize > maxBlobSize {
+		return nil, fmt.Errorf("blob size cannot exceed 2 MiB")
+	}
+	if blobSize == 0 {
+		return nil, fmt.Errorf("blob size must be greater than 0")
+	}
+
+	securityParams, err := s.tx.GetQuorumSecurityParams(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	seenQuorums := make(map[uint8]struct{})
+	// The quorum ID must be in range [0, 254]. It'll actually be converted
+	// to uint8, so it cannot be greater than 254.
+	for _, quorumID := range req.GetQuorums() {
+		if _, ok := seenQuorums[quorumID]; ok {
+			return nil, fmt.Errorf("invalid request: security_params must not contain duplicate quorum_id")
+		}
+		seenQuorums[quorumID] = struct{}{}
+
+		if quorumID >= s.quorumCount {
+			err := s.updateQuorumCount(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get onchain quorum count: %w", err)
+			}
+
+			if quorumID >= s.quorumCount {
+				return nil, fmt.Errorf("invalid request: the quorum_id must be in range [0, %d], but found %d", s.quorumCount-1, quorumID)
+			}
+		}
+	}
+
+	// Set first two quorums to be required
```

Comment:
We shouldn't introduce coupling just yet. I think all we do for now should be just remove security params from API

Reply:
I added a "TestMode" parameter which disables the enforcement. Why are you suggesting that we shouldn't introduce this yet?